src/main/scala/soc/ChipIdPin.scala

```diff
@@ -0,0 +1,44 @@
+package testchipip.soc
+
+import chisel3._
+import chisel3.util._
+import org.chipsalliance.cde.config._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.regmapper._
+import freechips.rocketchip.subsystem._
+
+case class ChipIdPinParams (
+  numChips: Int = 1, 
```

Comment:
Instead of `numChips`, lets just make this `width`. Everything is in relation to the width.